tests/unit_tests/lcls_tools/common/measurements/test_screen_beam_profile.py

```diff
@@ -0,0 +1,119 @@
+from lcls_tools.common.measurements.screen_beam_profile_measurement import (
+    ScreenBeamProfileMeasurement,
+)
+from lcls_tools.common.data.fit.projection import ProjectionFit
+from lcls_tools.common.data.fit.methods import GaussianModel
+from lcls_tools.common.image.processing import ImageProcessor
+from lcls_tools.common.image.roi import ROI
+from lcls_tools.common.devices.device import Metadata
+from lcls_tools.common.devices.screen import (
+    ScreenControlInformation,
+    ScreenPVSet,
+    Screen,
+)
+import numpy as np
+import unittest
+
+
+class ScreenTest(Screen):
+    @property
+    def image(self) -> np.ndarray:
+        return self._image
+
+    @image.setter
+    def image(self, image):
+        self._image = image
+
+
+class TestScreenBeamProfileMeasurement(unittest.TestCase):
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+        self.test_instantiate_pydantic_objects()
+
+    def create_test_image(self, size: tuple, center: list, radius: int):
+        # make img that is a circle in the center of the image with known
+        # standard dev and mean. no imports, no calls to external or
+        # internal files.
+        image = np.zeros(size)
+        for y in range(image.shape[0]):
+            for x in range(image.shape[1]):
+                distance = np.sqrt((x - center[0]) ** 2 + (y - center[1]) ** 2)
+                if distance < radius:
+                    image[y, x] = 1
+        return image
+
+    def test_instantiate_pydantic_objects(self):
+        # creating image processing class
+        self.radius = 50
+        self.size = (800, 800)
+        self.center = [400, 400]
+        self.extent = [300, 300]
```

Comment:
Many of these are unused(?). With this many unnecessary intermediate variables (why on self?), this function is hard to read. Put the mocked objects in a separate function so the intent (test) of this function is clear.